### PR TITLE
SALTO-6166: Avoid serializing redundant values

### PR DESCRIPTION
**Pull request by ori-moisis**
When serializing element IDs and Fields, only write the attributes that are needed to deserialize

---

_Additional context for reviewer_
On an example workspace (relatively small salesforce environment) - this reduced the overall size of the cache (rocksdb)  from 21MB to 14MB
most likely, this would mostly scale with the number of references in the workspace...

---
_Release Notes_: 
Core:
- Reduced size of state file and cache

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/workspace/src/serializer/elements.ts [hunk 1]
```diff
@@ -200,12 +201,14 @@ export const serializeStream = async <T = Element>({
 
   const referenceExpressionReplacer = (e: ReferenceExpression): ReferenceExpression & SerializedClass => {
     if (e.value === undefined || referenceSerializerMode === 'keepRef' || !isVariableExpression(e)) {
-      return saltoClassReplacer(e.createWithValue(undefined))
+      // eslint-disable-next-line no-use-before-define
+      return saltoClassReplacer(_.cloneDeepWith(e.createWithValue(undefined), replacer))
```
Comment (ori-moisis) [inline packages/workspace/src/serializer/elements.ts:205]: by not going over this with a replacer previously, the code for `elemIdReplacer` would not be called, and the element ID in the reference would be serialized with all of its members (including the `fullName`)

**Review by tamtamirr — APPROVED** [review]

File: packages/workspace/test/serializer/serializer.test.ts [hunk 2]
```diff
@@ -355,6 +357,16 @@ describe('State/cache serialization', () => {
         new StaticFile({ filepath: staticFile.filepath, hash: staticFile.hash }),
       )
     })
+    it('should serialize fields without values that are not used for deserialize', async () => {
```
Comment (Nurdok) [inline packages/workspace/test/serializer/serializer.test.ts:360]: The nittiest of nits: there's grammatical ambiguity here, I read this as "should serialize fields that don't have values". Can you rephrase to, e.g., "should drop values that are not use for deserialization while serializing fields" or something like that?

File: packages/workspace/src/serializer/elements.ts [hunk 3]
```diff
@@ -200,12 +201,14 @@ export const serializeStream = async <T = Element>({
 
   const referenceExpressionReplacer = (e: ReferenceExpression): ReferenceExpression & SerializedClass => {
     if (e.value === undefined || referenceSerializerMode === 'keepRef' || !isVariableExpression(e)) {
-      return saltoClassReplacer(e.createWithValue(undefined))
+      // eslint-disable-next-line no-use-before-define
```
Comment (Nurdok) [inline packages/workspace/src/serializer/elements.ts:204]: Can you avoid all these use-before-defines by wrapping these functions in a class?
Feel free to ignore

**Review by Nurdok — APPROVED** [review]

File: packages/workspace/src/serializer/elements.ts (hunk 3 above)
Comment (ori-moisis) [reply packages/workspace/src/serializer/elements.ts]: merging this for now, feel free to comment anyway, I'll open a separate refactor PR if we need to change this
